pkg/shell/superuser.jsx

```diff
@@ -203,10 +241,17 @@
     const [error, setError] = useState(null);
 
     const apply = () => {
+        const was_sudo = proxy.Current == "sudo";
         setError(null);
         proxy.Stop()
                 .then(() => {
-                    return cockpit.spawn(["sudo", "-k"], { host: host }).always(() => {
+                    function maybe_reset_sudo() {
+                        if (was_sudo)
+                            return cockpit.spawn(["sudo", "-k"], { host: host });
+                        else
+                            return Promise.resolve();
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17536-20220713-085711-6af71fce-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)